docs/ldml/tr35.html

```diff
@@ -1140,15 +1140,25 @@ <h4><a name="Canonical_Unicode_Locale_Identifiers" href="#Canonical_Unicode_Loca
       <li>Replace aliases in the <code><a href=
     "#unicode_language_id">unicode_language_id</a></code> and  tlang (if any) using the following process:
         <ul>
-          <li>If the language subtag matches the <i>type</i> attribute of a <i>languageAlias</i> element in <a href="tr35-info.html#Supplemental_Data">Supplemental
-            Data</a>, replace the language subtag with the <i>replacement</i> value.
-            <ol>
-              <li>If there are additional subtags in the <i>replacement</i> value, add them to the result, but
-                only if there is no corresponding subtag already in the
-                tag.</li>
-              <li>Five special deprecated grandfathered codes (such as <em>i-default</em>) are in  type attributes, and are also replaced.</li>
-            </ol>
+	  <li>Get the components of the <code><a href="#unicode_language_id">unicode_language_id</a></code> 
+		  (<i>language</i>, <i>script</i>, and <i>region</i>), plus any variants and extensions.</li>
+	  <li>For <i>trial</i> in {<i>language</i> _ <i>variant</i>,
+		  <i>language</i> _ <i>script</i> _ <i>region</i>,
```

Comment:
You always want the longest unit to match, so the items need to be tested in the following order:

 // 4 items
language_script_region_variants
// 3 items
language_script_region
language_script_variants
language_region_variants
// 2 items
language_script
language_region
language_variants
// 1 item
language

Note: the variants match only if all the variants in the source language_id match all of the variants in the languageAlias.

Reply:
could you explain a little bit about "the variants match only if all the variants in the source language_id match all of the variants in the languageAlias" by some example?